network/f5/bigip_gtm_datacenter.py

```diff
@@ -0,0 +1,569 @@
+
+    def set_location(self):
+        current = self.get_location()
+
+        if current != self._location:
+            self._payload = {
+                'location': self._location
+            }
+            return self.update()
+        else:
+            return False
+
+    def set_description(self):
+        current = self.get_description()
+
+        if current != self._description:
+            self._payload = {
+                'description': self._description
+            }
+            return self.update()
+        else:
+            return False
+
+    def present(self):
+        changed = False
+
+        if self.exists():
+            if self._contact is not None:
+                changed = self.set_contact()
+
+            if self._location is not None:
+                changed = self.set_location()
+
+            if self._description is not None:
+                changed = self.set_description()
+        else:
+            changed = self.create()
+
+        return changed
+
+    def absent(self):
+        changed = False
+
+        if self.exists():
+            changed = self.delete()
+
+        return changed
+
+    def enable(self):
+        changed = False
+        current = self.get_enabled()
+
+        if not current:
+            self.set_enabled()
+            changed = True
+
+        return changed
+
+    def disable(self):
+        changed = False
+        current = self.get_enabled()
+
+        if current:
+            self.set_disabled()
+            changed = True
+
+        return changed
+
+    def exists(self):
+        uri = '%s/%s' % ( self._uri, self._full_name)
+        resp = requests.get(uri,
+                            auth=(self._username, self._password),
+                            verify=self._validate_certs)
+        if resp.status_code == 200:
+            return True
+        else:
+            return False
+
+
+def main():
+    changed = False
+    icontrol = False
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            connection=dict(default='rest', choices=['icontrol', 'rest']),
+            contact=dict(required=False, default=None),
+            description=dict(required=False, default=None),
+            enabled=dict(type='bool'),
+            location=dict(required=False, default=None),
+            name=dict(require=True),
+            password=dict(default='admin'),
+            server=dict(required=True),
+            state=dict(choices=['present', 'absent']),
+            user=dict(default='admin'),
+            validate_certs=dict(default=True, type='bool')
+        )
+    )
+
+    connection = module.params.get('connection')
+    hostname = module.params.get('server')
+    password = module.params.get('password')
+    username = module.params.get('user')
+    state = module.params.get('state')
+    enabled = module.params.get('enabled')
+
+    try:
+        if connection == 'icontrol':
+            if not bigsuds_found:
+                pass
```

Comment:
Shouldn't this raise a problem if bigsuds isn't found? It's required for icontrol, isn't it?